terraform/modules/application/ecs.tf

```diff
@@ -94,20 +94,75 @@ resource "aws_ecs_task_definition" "app" {
   }
 }
 
+resource "aws_ecs_task_definition" "sidekiq" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-sidekiq"
+  cpu                      = var.ecs_sidekiq_task_cpu
+  execution_role_arn       = aws_iam_role.task_execution.arn
+  memory                   = var.ecs_sidekiq_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = aws_iam_role.task.arn
+
+  container_definitions = jsonencode([
+    {
+      name    = local.sidekiq_container_name
+      command = ["bundle", "exec", "sidekiq", "-t", "3"]
+      cpu     = var.ecs_sidekiq_task_cpu
+      environment = [
+        { Name = "EXPORT_PAAS_INSTANCE", Value = local.export_bucket_key },
+        { Name = "RAILS_ENV", Value = var.rails_env },
+        { Name = "REDIS_CONFIG", Value = "[{\"instance_name\":\"\",\"credentials\":{\"uri\":\"${var.redis_connection_string}\"}}]" },
+        { Name = "S3_CONFIG", Value = jsonencode(local.s3_config) }
+      ]
+      essential         = true
+      image             = var.ecr_repository_url
+      memoryReservation = var.ecs_sidekiq_task_memory * 0.75
+      user              = "nonroot"
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.this.id

```

Comment:
Potentially (maybe one log group for app, one for sidekiq and one for ad-hoc tasks). The group above was just to help debug ECS to get the app running.

But I don't think it should be done as part of this ticket but in the [logging epic](https://dluhcdigital.atlassian.net/browse/CLDC-2510). Which I noticed doesn't seem to have a ticket for ECS logging - shall I add this in @RachaelBooth?

Reply:
There is a ticket around log groups already, although it assumes tidying up the single current one rather than splitting them out. (It definitely does belong in that epic though).

My instinct is that given you are able to look at the logs per-container if you want to combining everything into one log group is fine (which gives the ability to search all of them at once), but I'm happy to hear other opinions on that.

Reply:
Ah I think that's ok if you can split out the logs per-container (haven't seen this before). If we find it's a hinderance having them all in one log group later we can always then split them out then